addons/survey/models/survey_question.py

```diff
@@ -710,35 +724,60 @@ class SurveyQuestionAnswer(models.Model):
     sequence = fields.Integer('Label Sequence order', default=10)
     scoring_type = fields.Selection(related='question_id.scoring_type')
     # answer related fields
-    value = fields.Char('Suggested value', translate=True, required=True)
+    value = fields.Char('Suggested value', translate=True)
     value_image = fields.Image('Image', max_width=1024, max_height=1024)
     value_image_filename = fields.Char('Image Filename')
+    value_label = fields.Char('Value Label', compute='_compute_value_label', translate=True,
+                              help="Answer label as either the value itself if not empty "
+                                   "or a letter representing the index of the answer otherwise.")
     is_correct = fields.Boolean('Correct')
     answer_score = fields.Float('Score', help="A positive score indicates a correct choice; a negative or null score indicates a wrong answer")
 
+    _sql_constraints = [
+        ('value_not_empty', "CHECK (value IS NOT NULL OR value_image_filename IS NOT NULL)",
+         'Suggested answer value must not be empty (a text and/or an image must be provided).'),
+    ]
+
     @api.depends('value', 'question_id.question_type', 'question_id.title', 'matrix_question_id')
     def _compute_display_name(self):
-        """Render an answer name as "Question title : Answer value", making sure it is not too long.
+        """Render an answer name as "Question title : Answer label value", making sure it is not too long.
 
         Unless the answer is part of a matrix-type question, this implementation makes sure we have
         at least 30 characters for the question title, then we elide it, leaving the rest of the
         space for the answer.
         """
         for answer in self:
             if not answer.question_id or answer.question_id.question_type == 'matrix':
-                answer.display_name = answer.value
+                answer.display_name = answer.value_label
                 continue
             title = answer.question_id.title or _("[Question Title]")
-            n_extra_characters = len(title) + len(answer.value) + 3 - self.MAX_ANSWER_NAME_LENGTH  # 3 for `" : "`
+            answer_label = answer.value_label or ''
```

Comment:
If it was just `label`, we wouldn't "need" this.

Reply:
as it is computed, how can it be False ? Maybe because it is not in the triggers ?